File: FrontTP9/src/types/Instrumento.ts
import Categoria from "./Categoria";

class Instrumento  {
    id:string = "0";
    instrumento: string = "0";
	marca:string = "0";
	modelo:string = "0";
    imagen: string = "0";
    precio: number = 0;
    costoEnvio: string = "0";
    cantidadVendida: number = 0;
	descripcion:string = "0";
    eliminado: boolean = false;
    categoria: Categoria = new Categoria();
};

export default Instrumento;

File: BackTP9/src/main/java/com/example/backtp4/model/Instrumento.java
package com.example.backtp4.model;

import com.fasterxml.jackson.annotation.JsonIgnore;
import jakarta.persistence.*;
import lombok.*;

import java.util.ArrayList;
import java.util.List;

@Entity
@Table(name = "instrumento")
@Getter
@Setter
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class Instrumento {
    @Id
    private String id;
    private String instrumento;
    private String marca;
    private String modelo;
    private String imagen;
    private Double precio;
    private String costoEnvio;
    private String cantidadVendida;
    private String descripcion;
    private boolean eliminado;
    @ManyToOne
    private CategoriaInstrumento categoria;
    @OneToMany(mappedBy = "instrumento")
    @JsonIgnore
    private List<PedidoDetalle> detalles = new ArrayList<>();
}
